canonicalwebteam/blog/common_view_logic.py

```diff
@@ -20,46 +26,48 @@ def get_index_context(page_param, articles, total_pages):
 
         category_ids = article["categories"]
 
+        # Can these calls be bundled?
+        first_item = True
         for category_id in category_ids:
             if category_id not in category_cache:
-                category_cache[category_id] = {}
-
+                try:
+                    resolved_category = api.get_category_by_id(category_id)
+                except Exception:
+                    resolved_category = None
+                category_cache[category_id] = resolved_category
+            if first_item:
+                article["display_category"] = category_cache[category_id]
```

Comment:
Do you think it's neater to roll these updates into `logic.transform_article`? It seems like kinda the same concern. It could take 2 extra arguments `fetch_categories=False` and `fetch_groups=False`. What do you think?

This would also help remove code duplication in the `get_article_context` function.

Reply:
Hm, they do have slightly different logic though. Since in `get_article_context` the loop breaks immediately.
I think this might be better as a new ticket?

Reply:
Filed here: https://github.com/canonical-webteam/blog/issues/21